SD_Card_Driver/Core/Src/LOS_D_sd.cpp

```diff
@@ -0,0 +1,96 @@
+#include "LOS_D_sd.hpp"
+
+int SDCard::init()
+{
+	return f_mount(&fs, "", 1);
+}
+
+int SDCard::open(const char* location)
+{
+	if (fileOpened && strcmp(location, filePath) == 0) {
+		return 0;
+	}
+
+	if (fileOpened) {
+		int res = f_close(&file);
+		if (res != 0) {
+			return res;
+		}
+	}
+
+	int res = f_open(&file, location, FA_OPEN_ALWAYS | FA_WRITE | FA_READ);

```

Comment:
It seems to me like we might want support to open more than one file at a time? is this possible or is it a limitation of the hal implementation? 

Is the idea that we would create multiple SDCard objects to perform such a task? Your use of the singleton design pattern suggests otherwise.

Reply:
The problem with opening multiple file at the same time is we need to keep track of them somehow.

The easiest way is to have an open function that opens a file and returns a file object, then the user would close the file with that file object later. However, different file IO libraries use different file objects. For example, `fopen` in C returns a different object from the `f_open` in the FatFS middleware. If we ever change the file IO library in the future, we might have to change the file object we use too, unless we have a generic file object type that can be redefined by each driver (not sure how to do this).

Another way is to keep track of all the file objects in an array/list within the driver, but this makes the driver much more complicated.

Reply:
re, generic file object type: this is something that should exist in LOS Core if I understand correctly.